This PR adds support for smart pointers

juanjqo opened the pull request:
@dqrobotics/developers 

Hi @mmmarinho, 

this PR adds support for smart pointers. This feature was adapted from `DQ_PseudoinverseController.h` and `DQ_KinematicController.h`.

The current example that use `LBR4pVrepRobot` and `YouBotVrepRobot` (vrep_interface_move_kuka_and_youbot.cpp) is working. Furthermore, I tested a version using smart pointers ([vrep_interface_move_kuka_and_youbot_smart_pointers.zip](https://github.com/dqrobotics/cpp-interface-vrep/files/10379439/vrep_interface_move_kuka_and_youbot_smart_pointers.zip)).  I'm going to improve the documentation in a future PR.

Best regards, 

Juancho

![](https://img.shields.io/badge/Tests-developer%20workflow-orange)![](https://img.shields.io/badge/Ubuntu%2022.04%20LTS%20(x64)-passing-passing)![](https://img.shields.io/badge/MacOS%2013.1%20(ARM64)%20-passing-passing)![](https://img.shields.io/badge/Tested%20on-CoppeliaSim%204.4-blue)

juanjqo committed:
[DQ_VrepRobot.h, cpp] Added support for smart pointers. Deprecated raw pointers.

juanjqo committed:
[DQ_VrepRobot.h, cpp] Fixed format of the header

juanjqo committed:
[LBR4pVrepRobot.cpp] Added support for smart pointers. Deprecated raw pointers. Added the _set_names method.

juanjqo committed:
[YouBotVrepRobot.h, cpp] Added support for smart pointers. Deprecated raw pointers, and added _set_names method.

mmmarinho commented:
@juanjqo  Thanks!

For this PR, just a reminder if needed, please add/update the doxygen docstrings for all constructors/methods you added/changed in this PR.

juanjqo committed:
[DQ_VrepRobot.cpp] Added documentation for new methods or constructors.

juanjqo committed:
[LBR4pVrepRobot.cpp, YouBotVrepRobot.cpp] Added documentation for new methods and constructors.

juanjqo committed:
[DQ_VrepRobot.cpp] Updated the documentation of the constructor

juanjqo committed:
This commit updates the documentation of the constructors

[DQ_VrepRobot.cpp,  LBR4pVrepRobot.cpp, YouBotVrepRobot.cpp] I used the same style used in DQ_SerialManipulatorDH.

juanjqo commented:
@mmmarinho I added the documentation for new methods/constructors.
